knimin/lib/tests/test_data_access.py

```diff
@@ -1451,6 +1462,60 @@ def _basic_test_steps_for_normalized_shotgun_plate(self, before, after,
 
             self.assertEqual(obs[k], exp_rnsp[k])
 
+    def test_generate_i5_i7_indexes(self):
+        # these tests will depend on what's in the db, which are added
+        # just below -- Add shotgun_index valid values
+
+        # generate 10 and check the values are just fine
+        obs = db.generate_i5_i7_indexes('iTru', 1)
+        exp = [('iTru5_01_B', 'iTru7_101_02')]
+        self.assertEqual(obs, exp)
+
+        # ask for another 10 and make sure it starts where we need to
+        obs = db.generate_i5_i7_indexes('iTru', 10)
+        exp = [('iTru5_01_C', 'iTru7_101_03'), ('iTru5_01_D', 'iTru7_101_04'),
+               ('iTru5_01_E', 'iTru7_101_05'), ('iTru5_01_F', 'iTru7_101_06'),
+               ('iTru5_01_G', 'iTru7_101_07'), ('iTru5_01_H', 'iTru7_101_08'),
+               ('iTru5_02_A', 'iTru7_101_09'), ('iTru5_02_B', 'iTru7_101_10'),
+               ('iTru5_02_C', 'iTru7_101_11'), ('iTru5_02_D', 'iTru7_101_12')]
+        self.assertEqual(obs, exp)
+
```

Comment:
Maybe per comment above, add some tests that verify that the number of actual unique barcode combinations returned is = to the expected maximum set size?